test_suite/tests/json.rs

```diff
@@ -171,9 +171,9 @@ fn test_tuplestruct() {
         "def": {
             "composite": {
                 "fields": [
-                    { "type": 1, "displayName": ["i32"] },
-                    { "type": 2 },
-                    { "type": 4, "displayName": ["bool"] },
+                    { "type": 1, "typeName": "i32" },
+                    { "type": 2, "typeName": "[u8 ; 32]" },
```

Comment:
This is nice. :)

Reply:
Note sure we need `displayName` or `typeName` (why rename btw?) for types such as `[u8; 32]` because its primary use case is still to counterfeit type aliases and besides that I cannot think of any other serious use case. Also I am unsure about turning the namespace (array of strings) into a single string since that kinda dictates how 3rd party tools handle namespaces or simply removes namespaces which are kinda important to differentiate between types.

Reply:
> displayName or typeName (why rename btw?)

To make it more general to imply we are not just dealing with paths/aliases now, but whatever type name is in that position.

> Note sure we need displayName or typeName (why rename btw?) for types such as [u8; 32]

Indeed it is redundant because you can just look it up on the underlying type. We can easily filter those out, but in this iteration I have decided to always take the type name in the source for simplicity's sake, at the cost of some redundancy. 

> besides that I cannot think of any other serious use case

I believe we might need this initially if we want to be able to convert to the existing substrate metadata format, though of course it may be possible without, needs some investigation.

>  Also I am unsure about turning the namespace (array of strings) into a single string since that kinda dictates how 3rd party tools handle namespaces or simply removes namespaces which are kinda important to differentiate between types.

3rd party tools should not be relying on any namespace in the type/display name because it will only be there if the user decides to fully qualify their type e.g. `struct S { field: foo::bar::Foo }`, if they just `use foo::bar::Foo` then the display/type name will be `Foo`. 

By not using the `Path` type anymore we are just giving an extra indication that this field does *not* provide any guarantees as to the content of that field other than it is the name of a valid Rust type in the context of the `.rs` file where it is defined.